tasks/image-segmentation/src/validation.py

```diff
@@ -0,0 +1,184 @@
+    for index, mask in enumerate(predictedMasks):
+        vPadding = testMasksPadding[index][0]
+        hPadding = testMasksPadding[index][1]
+
+        if vPadding > 0:
+            mask[:vPadding, :] = 0
+            mask[-vPadding:, :] = 0
+        if hPadding > 0:
+            mask[:, :hPadding] = 0
+            mask[:, -hPadding:] = 0
+
+    return predictedMasks
+
+
+def run(model: tf.lite.Interpreter, batchImages: np.ndarray) -> np.ndarray:
+    inputDetails = model.get_input_details()
+    outputDetails = model.get_output_details()
+
+    model.resize_tensor_input(inputDetails[0]["index"], batchImages.shape)
+    model.allocate_tensors()
+    model.set_tensor(inputDetails[0]["index"], batchImages.astype(np.float32))
+    model.invoke()
+
+    prediction = model.get_tensor(outputDetails[0]["index"])
+    prediction = np.argmax(prediction, axis = -1)
+
+    return prediction
+
+
+def predictionPrepare(batchSamples: list[ImageSample], imageSize: int, classes: ImageDatasetClasses) -> tuple[np.ndarray, list[np.ndarray], list[tuple[int, int]]]:
+    testImgs: list[np.ndarray] = []
+    testMasks: list[np.ndarray] = []
+    testMasksPadding: list[tuple[int, int]] = []
+
+    for sample in batchSamples:
+        sample.download()
+        sample.unzip()
+        sampleData = sample.load()
+
+        if hasDotAnnotation(sampleData.annotation):
+            logging.warning(f">> [Image Segmentation] Sample \"{sample.name}\" (ID: {sample.id}) has invalid annotation (too few coordinates). Skipping Sample")
+            continue
+
+        resized, _, _ = resizeWithPadding(sampleData.image, imageSize, imageSize)
+        normalized = resized / 255
+
+        groundtruth = sampleData.extractSegmentationMask(classes)
+        groundtruth, vPadding, hPadding = resizeWithPadding(groundtruth, imageSize, imageSize)
+
+        testImgs.append(normalized)
+        testMasks.append(groundtruth)
+        testMasksPadding.append((vPadding, hPadding))
+
+    testImages = np.array(testImgs)
+
+    return (testImages, testMasks, testMasksPadding)
+
+
+def loadModel(modelPath: Path) -> tf.lite.Interpreter:
+    modelInterpreter = tf.lite.Interpreter(str(modelPath / "model.tflite"))
+    return modelInterpreter
+
+
+def validation(taskRun: TaskRun) -> None:
+    dataset = taskRun.dataset
+    imageSize: int = taskRun.parameters["imageSize"]
+    batchSize: int = taskRun.parameters["batchSize"]
+    dataset.classes.exclude(taskRun.parameters["excludedClasses"])
+
+    model: Model = taskRun.parameters["trainedModel"]
+    model.download()
+    modelInterpreter = loadModel(model.path)
+
+    iouScores: list[float] = []
+    fieldNamesSamples = ["Sample ID", "Sample Name", "IoU Score", "Accuracy"]
+    csvSamplesData: list[dict[str, str]] = []
+
+    for startIndex in range(0, dataset.count, batchSize):
+        batchSamples = dataset.samples[startIndex : startIndex + batchSize]
```

Comment:
Can you remove the whitespace around `:`, as that's a standard when slicing lists. This way it will also be consistent with list slicing on lines 78 to 82